### PR TITLE
add self shutdown route example

### PR DESCRIPTION
**Pull request by robjtede**
in response to actix/actix-web#1315

### CODE REVIEW

**Comment by JohnTitor** [comment]
It'd be nice if the example had README with a brief overview and a link to the issue.

**Comment by Dowwie** [comment]
I realize this was in line with the original request but the thought of having the power to shut down a server via an http request sends chills down my spine.  What do you think about recommending against this in the README and instead using local signal handling?

**Comment by JohnTitor** [comment]
:+1:

File: self-shutdown-route/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            .data(stopper)
+            .wrap(middleware::Logger::default())
+            .service(hello)
+            .service(stop)
+    })
+    .bind(&bind)?
+    .run();
+
+    // clone the Server handle
+    let srv = server.clone();
```
Comment (azadsagar) [inline self-shutdown-route/src/main.rs:45]: Why do I get : no method named `clone` found for struct `actix_web::dev::Server` in the current scope ?

Comment (robjtede) [reply self-shutdown-route/src/main.rs:45]: because this example has since been reworked for actix web v4, see the master branch's shutdown-server example

**Pull request merged**